### PR TITLE
podvm: update RHEL and CentOS 

### PR DESCRIPTION
**Pull request by snir911**

### CODE REVIEW

File: podvm/Dockerfile.podvm_builder.centos [hunk 1]
```diff
@@ -19,12 +19,11 @@ ENV GO_VERSION ${GO_VERSION}
 ENV RUST_VERSION ${RUST_VERSION}
 ENV PROTOC_VERSION ${PROTOC_VERSION}
 
-RUN dnf install -y 'dnf-command(config-manager)' && dnf config-manager --set-enabled powertools && \
-    dnf install -y epel-release epel-next-release && dnf update -y && \
+RUN dnf install -y 'dnf-command(config-manager)' && dnf config-manager --set-enabled crb && \
     dnf groupinstall -y 'Development Tools' && \
-    dnf install -y yum-utils gnupg git curl pkg-config libseccomp-devel gpgme-devel \
+    dnf install -y yum-utils gnupg git pkg-config libseccomp-devel gpgme-devel \
```
Comment (bpradipt) [inline podvm/Dockerfile.podvm_builder.centos:24]: curl is not needed ?

Comment (snir911) [reply podvm/Dockerfile.podvm_builder.centos:24]: it seems it's already shipped, trying to reinstall it hits a conflict:
```
podman run --rm quay.io/centos/centos:stream9 dnf install curl
CentOS Stream 9 - BaseOS                        2.2 MB/s | 7.9 MB     00:03    
CentOS Stream 9 - AppStream                     4.8 MB/s |  18 MB     00:03    
CentOS Stream 9 - Extras packages               9.3 kB/s |  14 kB     00:01    
Error: 
 Problem: problem with installed package curl-minimal-7.76.1-26.el9.x86_64
  - package curl-minimal-7.76.1-26.el9.x86_64 from @System conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-20.el9.x86_64 from baseos conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-21.el9.x86_64 from baseos conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-22.el9.x86_64 from baseos conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-23.el9.x86_64 from baseos conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-26.el9.x86_64 from baseos conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-12.el9.x86_64 from appstream conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - package curl-minimal-7.76.1-14.el9.x86_64 from appstream conflicts with curl provided by curl-7.76.1-26.el9.x86_64 from baseos
  - cannot install the best candidate for the job
(try to add '--allowerasing' to command line to replace conflicting packages or '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

File: podvm/qcow2/centos/variables.pkr.hcl [hunk 2]
```diff
@@ -21,12 +21,12 @@ variable "disk_size" {
 
 variable "cloud_image_checksum" {
   type    = string
-  default = "b80ca9ccad8715aa1aeee582906096d3190b91cebe05fcb3ac7cb197fd68a7fe"
+  default = "d12bb6934dd207e242d6aa13f6a4ca4969449c14c3bbdd88a5ce5f5203597a40"
```
Comment (wainersm) [inline podvm/qcow2/centos/variables.pkr.hcl:24]: Not for this PR, wondering if it is possible to read the checksum from `versions.yaml`, is it?

Comment (snir911) [reply podvm/qcow2/centos/variables.pkr.hcl:24]: I thought we may be able to just drop the defaults instead, need to verify it doesn't break anything, i thought to do it later

File: Makefile.defaults [hunk 3]
```diff
@@ -17,14 +17,17 @@ $(or $(shell $(YQ_COMMAND) '.$(1) | select(. != null)' $(VERSIONS_SRC)), \
...
+centos_RELEASE ?= stream-9
+rhel_RELEASE ?= 9.2
 
 ubuntu_amd64_IMAGE_URL := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).amd64.url)
 ubuntu_amd64_IMAGE_CHECKSUM := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).amd64.checksum)
 ubuntu_s390x_IMAGE_URL := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).s390x.url)
 ubuntu_s390x_IMAGE_CHECKSUM := $(call query,cloudimg.ubuntu.$(ubuntu_RELEASE).s390x.checksum)
 centos_amd64_IMAGE_URL := $(call query,cloudimg.centos.$(centos_RELEASE).amd64.url)
 centos_amd64_IMAGE_CHECKSUM := $(call query,cloudimg.centos.$(centos_RELEASE).amd64.checksum)
+rhel_amd64_IMAGE_URL := $(call query,cloudimg.centos.$(centos_RELEASE).amd64.url)
```
[the remark this answers is not in the document]
Comment (snir911) [reply Makefile.defaults:29]: good catch

File: podvm/qcow2/centos/variables.pkr.hcl (hunk 2 above)
Comment (wainersm) [reply podvm/qcow2/centos/variables.pkr.hcl:24]: cool, it would be nice to remove that if it is a simple duplication of code.

**Comment by wainersm** [comment]
@snir911 I tried the previous version of this PR, it failed to build the podvm because it was trying to pull the CentOS Stream 8 image still:

```
[wmoschet@wainer-laptop podvm]$ docker build -t podvm_centos --build-arg BUILDER_IMG=podvm_builder_centos --build-arg BINARIES_IMG=podvm_binaries_centos -f Dockerfile.podvm.centos .
[+] Building 6.4s (12/13)                                                                                                docker:default
 => [internal] load build definition from Dockerfile.podvm.centos                                                                  0.1s
 => => transferring dockerfile: 2.21kB                                                                                             0.0s
 => [internal] load .dockerignore                                                                                                  0.1s
 => => transferring context: 2B                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/podvm_builder_centos:latest                                                     0.0s
 => [internal] load metadata for docker.io/library/podvm_binaries_centos:latest                                                    0.0s
 => CACHED [podvm_builder 1/7] FROM docker.io/library/podvm_builder_centos                                                         0.0s
 => CACHED [podvm_binaries 1/1] FROM docker.io/library/podvm_binaries_centos                                                       0.0s
 => [podvm_builder 2/7] RUN if [ -n "" ]; then        rm -rf cloud-api-adaptor &&        git clone  cloud-api-adaptor;     fi &&   0.8s
 => [podvm_builder 3/7] COPY --from=podvm_binaries /podvm-binaries.tar.gz /src/cloud-api-adaptor/podvm/files                       0.2s
 => [podvm_builder 4/7] RUN tar xvf /src/cloud-api-adaptor/podvm/files/podvm-binaries.tar.gz -C /src/cloud-api-adaptor/podvm/file  1.5s
 => [podvm_builder 5/7] COPY --from=podvm_binaries /pause-bundle.tar.gz /src/cloud-api-adaptor/podvm/files                         0.2s 
 => [podvm_builder 6/7] RUN tar xvf /src/cloud-api-adaptor/podvm/files/pause-bundle.tar.gz -C /src/cloud-api-adaptor/podvm/files   0.3s 
 => ERROR [podvm_builder 7/7] RUN cd cloud-api-adaptor/podvm &&      make BINARIES=  PAUSE_BUNDLE= image                           3.0s 
------                                                                                                                                  
 > [podvm_builder 7/7] RUN cd cloud-api-adaptor/podvm &&      make BINARIES=  PAUSE_BUNDLE= image:                                      
0.403 rm -fr output                                                                                                                     
0.404 rm -f cloud-init.img                                                                                                              
0.405 cloud-localds cloud-init.img qcow2/userdata.cfg                                                                                   
0.519 mkdir -p toupload                                                                                                                 
0.520 if [ "0" = "1" ] && [ "x86_64" = "s390x" ]; then \
0.520 	qemu-img create -f qcow2 "se-podvm-cc34a12-dirty-amd64.qcow2" 100G; \
0.520 fi
0.522 packer build -var qemu_image_name=podvm-cc34a12-dirty-amd64.qcow2 -var cloud_image_url=https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-GenericCloud-8-20220913.0.x86_64.qcow2 -var cloud_image_checksum=sha256:8717251f8e4d2fe3e5032799caae89358c1ba68d65a16b5128a59ec6003aac1c  qcow2/centos
0.686 Warning: Bundled plugins used
0.686 
0.686 This template relies on the use of plugins bundled into the Packer binary.
0.686 The practice of bundling external plugins into Packer will be removed in an
0.686 upcoming version.
0.686 
0.686 To remove this warning, add the following section to your template:
0.686 
0.686 packer {
0.686   required_plugins {
0.686     qemu = {
0.686       source  = "github.com/hashicorp/qemu"
0.686       version = "~> 1"
0.686     }
0.686   }
0.687 }
0.687 
0.687 Then run 'packer init' to manage installation of the plugins
0.687 
0.687 
0.878 qemu.centos: output will be in this color.
0.878 
0.879 ==> qemu.centos: Retrieving ISO
0.879 ==> qemu.centos: Trying https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-GenericCloud-8-20220913.0.x86_64.qcow2
0.880 ==> qemu.centos: Trying https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-GenericCloud-8-20220913.0.x86_64.qcow2?checksum=sha256%3A8717251f8e4d2fe3e5032799caae89358c1ba68d65a16b5128a59ec6003aac1c
2.314 ==> qemu.centos: Download failed bad response code: 404
2.314 ==> qemu.centos: error downloading ISO: [bad response code: 404]
2.315 Build 'qemu.centos' errored after 1 second 436 milliseconds: error downloading ISO: [bad response code: 404]
2.315 
2.315 ==> Wait completed after 1 second 436 milliseconds
2.315 
2.315 ==> Some builds didn't complete successfully and had errors:
2.315 --> qemu.centos: error downloading ISO: [bad response code: 404]
2.315 
2.316 ==> Builds finished but no artifacts were created.
2.957 make: *** [Makefile:78: podvm-cc34a12-dirty-amd64.qcow2] Error 1
------
Dockerfile.podvm.centos:58
--------------------
  57 |     
  58 | >>> RUN cd cloud-api-adaptor/podvm && \
  59 | >>>      make BINARIES=  PAUSE_BUNDLE= image
  60 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c cd cloud-api-adaptor/podvm &&      make BINARIES=  PAUSE_BUNDLE= image" did not complete successfully: exit code: 2
```

**Comment by snir911** [comment]
@wainersm i think you should point CAA and CAA_REF to my branch

**Review by bpradipt — APPROVED** [review]
/lgtm
Not tested though.

**Comment by snir911** [comment]
s/9.2/9 because it cannot parse "."

**Comment by snir911** [comment]
@wainersm try again, apparently centOS9 requires x86 v2 features, i set default machine type which supports it.
(it would have work also with host, but it seems we don't set kvm by default so it doesn't work)

**Comment by wainersm** [comment]
@snir911 worked out!

```
wmoschet@wainer-laptop podvm]$ docker build -t podvm_centos_amd64 -f Dockerfile.podvm.centos --build-arg CAA_SRC=https://github.com/snir911/cloud-api-adaptor --build-arg CAA_SRC_REF=bump_rhel_centos --build-arg BUILDER_IMG=docker.io/library/podvm_builder_centos --build-arg BINARIES_IMG=docker.io/library/podvm_binaries_centos_amd64 .
[+] Building 480.7s (14/14) FINISHED                                                                                     docker:default
 => [internal] load build definition from Dockerfile.podvm.centos                                                                  0.1s
 => => transferring dockerfile: 2.21kB                                                                                             0.0s
 => [internal] load .dockerignore                                                                                                  0.0s
 => => transferring context: 2B                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/podvm_builder_centos:latest                                                     0.0s
 => [internal] load metadata for docker.io/library/podvm_binaries_centos_amd64:latest                                              0.0s
 => CACHED [podvm_builder 1/7] FROM docker.io/library/podvm_builder_centos                                                         0.0s
 => [podvm_binaries 1/1] FROM docker.io/library/podvm_binaries_centos_amd64                                                        0.2s
 => [podvm_builder 2/7] RUN if [ -n "https://github.com/snir911/cloud-api-adaptor" ]; then        rm -rf cloud-api-adaptor &&      4.7s
 => [podvm_builder 3/7] COPY --from=podvm_binaries /podvm-binaries.tar.gz /src/cloud-api-adaptor/podvm/files                       0.2s
 => [podvm_builder 4/7] RUN tar xvf /src/cloud-api-adaptor/podvm/files/podvm-binaries.tar.gz -C /src/cloud-api-adaptor/podvm/file  1.1s 
 => [podvm_builder 5/7] COPY --from=podvm_binaries /pause-bundle.tar.gz /src/cloud-api-adaptor/podvm/files                         0.2s 
 => [podvm_builder 6/7] RUN tar xvf /src/cloud-api-adaptor/podvm/files/pause-bundle.tar.gz -C /src/cloud-api-adaptor/podvm/files   0.4s 
 => [podvm_builder 7/7] RUN cd cloud-api-adaptor/podvm &&      make BINARIES=  PAUSE_BUNDLE= image                               465.0s 
 => [stage-2 1/1] COPY --from=podvm_builder /src/cloud-api-adaptor/podvm/output/*.qcow2 /                                          0.9s 
 => exporting to image                                                                                                             3.6s 
 => => exporting layers                                                                                                            3.6s 
 => => writing image sha256:b9665e1eb20ee9887a4f5e9057884d0dd63f175726fb2e201daf5753eb7002b9                                       0.0s 
 => => naming to docker.io/library/podvm_centos_amd64
```

**Review by wainersm — APPROVED** [review]
LGTM. I tested only CentOS podvm build, though.

**Comment by wainersm** [comment]
Re-tested latest updates, still building

**Pull request merged**